UnleashedRecomp/user/persistent_data.cpp

```diff
@@ -0,0 +1,18 @@
+#include "persistent_data.h"
+
+bool PersistentData::VerifySignature() const
+{
+    char sig[4] = EXT_SIGNATURE;
+
+    return memcmp(Header.Signature, sig, sizeof(Header.Signature)) == 0;
+}
+
+bool PersistentData::VerifyVersion() const
+{
+    return Header.Version == ExtVersion EXT_VERSION;
```

Comment:
Version checks should fail if the version is above the known version, not if it doesn't match. Lower versions should be handled by the parser instead.